recipes/celero/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "6",
+            "Visual Studio": "14",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("celero requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("celero requires C++14, which your compiler does not support.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("Celero-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmakelists,
+                              "target_compile_options(${PROJECT_NAME} PRIVATE -fPIC)",
+                              "")
+        tools.replace_in_file(cmakelists,
+                              "ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/static",
+                              "ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CELERO_COMPILE_DYNAMIC_LIBRARIES"] = self.options.shared
+        self._cmake.definitions["CELERO_ENABLE_EXPERIMENTS"] = False
+        self._cmake.definitions["CELERO_ENABLE_FOLDERS"] = False
+        self._cmake.definitions["CELERO_ENABLE_TESTS"] = False
+        self._cmake.definitions["CELERO_TREAT_WARNINGS_AS_ERRORS"] = False
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("license.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        # FIXME: official CMake target is exported without namespace
```

Comment:
```suggestion
        # FIXME: official CMake target is exported without namespace
        self.cpp_info.filenames["cmake_find_package"] = "Celero"
        self.cpp_info.filenames["cmake_find_package_multi"] = "Celero"
```

https://github.com/DigitalInBlue/Celero/tree/v2.6.0/cmake

Reply:
Yes actually, celero:
- has a module file `FindCelero.cmake` (without targets)
- exports a target `celero` in a file but without the config file, so it's useless :p